### ISSUE TITLE
TypeError when using Resolve as Pattern Generator

### ISSUE THREAD

**Issue by Fe-Lit** [eoyilmaz/displaycal-py3#148]
Hello you great people,
there is an issue when using Davinci Resolve as a pattern generator. OS is macOS 12.4 on an Intel mac.
Here is the log from the console:

```shell
DisplayCAL: Setting ARGYLL_MIN_DISPLAY_UPDATE_DELAY_MS 1000
DisplayCAL: Sending RGB 0.500 0.500 0.500, background RGB 0.123 0.123 0.123, x 0.2768, y 0.1652, w 0.4007, h 0.6411
TypeError: TypeError("a bytes-like object is required, not 'str'")
┌──────────────────────────────────────────────────────────────────────────────┐
│ /usr/local/bin/dispread -C "/usr/bin/env python" ./.wait -v -K               │
│ /Users/THEUSER/displaycal_venv/lib/python3.9/site-packages/DisplayCAL/linear.cal │
│ -ddummy -c1 -yn 0_16                                                         │
│ Traceback (most recent call last):                                           │
│   File                                                                       │
│ "/Users/THEUSER/displaycal_venv/lib/python3.9/site-packages/DisplayCAL/worker.py", │
│ line 7031, in exec_cmd                                                       │
│     self.setup_patterngenerator(logfiles)                                    │
│   File                                                                       │
│ "/Users/THEUSER/displaycal_venv/lib/python3.9/site-packages/DisplayCAL/worker.py", │
│ line 12983, in setup_patterngenerator                                        │
│     self.patterngenerator_send(                                              │
│   File                                                                       │
│ "/Users/THEUSER/displaycal_venv/lib/python3.9/site-packages/DisplayCAL/worker.py", │
│ line 13075, in patterngenerator_send                                         │
│     self.patterngenerator.send(rgb, bgrgb, x=x, y=y, w=w, h=h)               │
│   File                                                                       │
│ "/Users/THEUSER/displaycal_venv/lib/python3.9/site-packages/DisplayCAL/patterngenerators.py", │
│ line 536, in send                                                            │
│     self.conn.sendall("%s%s" % (struct.pack(">I", len(xml)), xml))           │
│ TypeError: a bytes-like object is required, not 'str'                        │
└──────────────────────────────────────────────────────────────────────────────┘
DisplayCAL: Sending RGB 0.500 0.500 0.500, background RGB 0.123 0.123 0.123, x 0.2768, y 0.1652, w 0.4007, h 0.6411
┌──────────────────────────────────────────────────────────────────────────────┐
│ a bytes-like object is required, not 'str'                                   │
└──────────────────────────────────────────────────────────────────────────────┘
```

**Comment by p5k369** [comment]
Hey @Fe-Lit, 
take a look at #149. I can not test the code atm but socket.sendall definitly expects bytes instead of str so this might do the trick.
You can try it out by changing your code locally.

**Comment by eoyilmaz** [comment]
sorry guys for my late response, this should be fixed with 409be4d

**Comment by Fe-Lit** [comment]
Ay, that was quick, awesome! Im going to test it further when im in the shop again next week.
